app/views/result/index.html.erb

```diff
@@ -3,157 +3,185 @@
-    <% end %>
-    <% url += "&count=" + count %>
-
-
-    <%# Creates an empty hash to put the search results from our 'A'
-        test in and a counter to increment within each of the following
-        loops                           %>
-    <% a_list = {} %>
-    <% a_counter = 0 %>
-    <% b_counter = 0 %>
-    <div class="ab-search-wrapper-left">
-      <h2 class="results-header">A Results</h2>
-
-      <%# Adds the A test parameter to the URL  %>
-      <% uri = URI("#{url}&ab_tests=#{params["which_test"]}:A") %>
-
-      <%# Gets a result from the search API based on the URL we made %>
-      <% response = Net::HTTP.get(uri) %>
-      <% findings = JSON.parse(response) %>
-
-      <%# Loops over the 'results' section of our result %>
-      <% findings["results"].each do |each_bit| %>
-
-        <%# increments counter and creates hash entry based on link and counter %>
-        <% a_counter += 1 %>
-        <% a_list["#{each_bit["link"]}"] = a_counter %>
-
-        <%# displays this in the 'A' section %>
-        <div class="changeless-box">
-          <% link = "https://gov.uk" + each_bit["link"] %>
-          <a href = "<%= link %>" ><%= each_bit["title"] %></a>
-        </div>
-        <br><br>
+    <% results = Searching.new(params).call %>
+    <table class="results-table" >
+      <tr class="table-row" class="table-row">
+        <th></th>
+        <th scope="col"><h3 class="heading-27" >A Results</h3></th>
+        <th scope="col"><h3 class="heading-27" >B Results</h3></th>
+        <th scope="col"><h3 class="heading-27" >Difference</h3></th>
+      </tr>
+      <% results.results.each do |result| %>
+        <tr class="table-row">
+
+          <th >
+            <p class="table-body table-column-with-padding">
+              <%= "#{result.position + 1 } -"%>
+            </p>
+          </th>
+
+          <% sides = [result.left, result.right] %>
+          <% sides.each do |side| %>
+            <th class="results-column">
+              <p class="results-text" >
+                <a href = "<%= "#{result.link(side)}" %>" >
+                  <%= truncate(result.name(side), :length => 60 ) %>
+                </a>
+                <p><%= result.title(side) %></p><br>
+                <span class="head-info">
+
+                  <% head_info_list = [] %>
+                  <% head_info_list << result.doc_format(side).split("_").join(" ").capitalize << result.date(side) %>
+                  <% if params["info"] == "enhanced" && params["historical"] == "on" ; head_info_list << result.historical(side) ; end %>
+                  <% if params["info"] == "enhanced" && params["popularity"] == "on" ; head_info_list << "Popularity: #{result.popularity(side)}" ; end %>
+                  <%= head_info_list.compact.join(", ") %>
+                </span>
+                <br>
+                <span class="head-info">
+                  <% if params["info"] == "enhanced" && params["content_id"] == "on" && result.content_id(side) != nil %>
```

Comment:
This flagging structure feels overly complicated given you are checking three values each time it is used, can we simplify this or possibly remove it and just make it the default if it is important.